client/client.go

```diff
@@ -259,6 +265,36 @@ func (c *Client) Pay(channelId types.Destination, amount *big.Int) {
 	c.engine.PaymentRequestsFromAPI <- engine.PaymentRequest{ChannelId: channelId, Amount: amount}
 }
 
+// CreatePayment returns (via a go channel) a voucher that increases the redeemable balance
+// of the given payment channel by the given amount.
+//
+// The returned voucher must be sent to the payee.
+func (c *Client) CreatePayment(channelId types.Destination, amount *big.Int) (<-chan payments.Voucher, error) {
+	vChan := make(chan payments.Voucher, 1)
+	id := rand.Uint64()
+	c.engine.PaymentRequestsFromAPI <- engine.PaymentRequest{ChannelId: channelId, Amount: amount, Withhold: true, PaymentID: id}
```

Comment:
Instead of routing the payment request to the engine I wonder if `CreatePayment` could call `VoucherManager.Pay` directly. That would mean the node has to have a reference to the `VoucherManager` (and the voucher manager would have to be concurrency-safe) but this would avoid some of the pain of routing through the engine and adding a `Withold` flag.

Instead we'd just call `VoucherManager.Pay` to generate a new Voucher and return that. Since we'd avoid the engine we'd avoid sending the voucher over the message service.